outros-exercicios/livraria/src/main/java/com/example/livraria/service/AluguelService.java

```diff
@@ -0,0 +1,139 @@
+import com.example.livraria.model.Locatario;
+import com.example.livraria.repository.AluguelRepository;
+import com.example.livraria.repository.LivroRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class AluguelService {
+
+    @Autowired
+    private AluguelRepository aluguelRepository;
+
+    @Autowired
+    private LivroRepository livroRepository;
+
+    public void setLocatario(Locatario locatario, Aluguel aluguel) {
+        if(locatario == null)
+            throw new IllegalArgumentException("Locatário inválido");
+
+        aluguel.setLocatario(locatario);
+    }
+
+    public void setLivros(List<Livro> livros, Aluguel aluguel) {
+        if(livros == null || livros.isEmpty())
+            throw new IllegalArgumentException("Lista de livros inválida");
+        else
+            aluguel.setLivros(livros);
+    }
+
+    public void setDataRetirada(LocalDate dataRetirada, Aluguel aluguel) {
+        if(dataRetirada == null)
+            throw new IllegalArgumentException("Data de retirada inválida");
+
+        aluguel.setDataRetirada(dataRetirada);
+    }
+
+    public void setDataDevolucao(LocalDate dataDevolucao, Aluguel aluguel) {
+        if(dataDevolucao == null)
+            throw new IllegalArgumentException("Data de devolução inválida");
+
+        aluguel.setDataDevolucao(dataDevolucao);
+    }
+
+    public List<AluguelDto> listarTodos() {
+        List<Aluguel> aluguelLista = aluguelRepository.findAll();
+        List<AluguelDto> aluguelDtoLista = new ArrayList<>();
+        for(Aluguel aluguel : aluguelLista) {
+            AluguelDto aluguelDto = new AluguelDto();
+            aluguelDto.setId(aluguel.getId());
+            aluguelDto.setLocatario(aluguel.getLocatario());
+            aluguelDto.setLivros(aluguel.getLivros());
+            aluguelDto.setDataRetirada(aluguel.getDataRetirada());
+            aluguelDto.setDataDevolucao(aluguel.getDataDevolucao());
+            aluguelDtoLista.add(aluguelDto);
+        }
+        return aluguelDtoLista;
+    }
+
+    public AluguelDto buscarPorId(Long id) {
+        Optional<Aluguel> aluguelOptional = aluguelRepository.findById(id);
+        if(aluguelOptional.isPresent()) {
+            Aluguel aluguel = aluguelOptional.get();
+            AluguelDto aluguelDto = new AluguelDto();
+            aluguelDto.setId(aluguel.getId());
+            aluguelDto.setLocatario(aluguel.getLocatario());
+            aluguelDto.setLivros(aluguel.getLivros());
+            aluguelDto.setDataRetirada(aluguel.getDataRetirada());
+            aluguelDto.setDataDevolucao(aluguel.getDataDevolucao());
+            return aluguelDto;
+        } else
+            throw new IllegalArgumentException("Aluguel não encontrado");
+    }
+
+    public AluguelDto adicionar(AluguelDto aluguelDto) {
```

Comment:
eu nao entendi direito o que esse método faz. Pra mim se for para salvar o aluguel no banco, seria apenas transformar o dto e retornar um ResponseEntity<Aluguel>